### DISCUSSION TITLE
How to refresh script module state on click, like how pulseaudio module does on click?

### DISCUSSION THREAD

**Discussion by amithm7** [polybar/polybar#2884] [Q&A]
I have this script module for mike:

```ini
[module/mike]
type = custom/script
exec = echo $(amixer | grep 'Capture') | rev | cut -d" " -f1 | rev | sed 's/\[//' | sed 's/\]//'
label = %output%
format-prefix = "MIC "
format-prefix-foreground = ${colors.primary}
click-left = pactl set-source-mute @DEFAULT_SOURCE@ toggle
```

By default, it has an interval of 5, I could set interval to 1, but it is still not fast enough.

Is there any way to refresh on click?

**Comment by patrick96** [comment]
The [wiki](https://github.com/polybar/polybar/wiki/Module:-script#examples) has an example for a script that changes its state when clicked. But crucially, it also updates on click. It is a bit more involved though because you have to manually create the update loop
